src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1654,4 +1662,101 @@ private function doExport($value)
 
         return $export;
     }
+
+    private function generateSignature(\ReflectionFunctionAbstract $r)
+    {
+        $signature = array();
+
+        foreach ($r->getParameters() as $p) {
+            $k = '$'.$p->name;
+            if (method_exists($p, 'isVariadic') && $p->isVariadic()) {
+                $k = '...'.$k;
+            }
+            if ($p->isPassedByReference()) {
+                $k = '&'.$k;
+            }
+            if (method_exists($p, 'getType')) {
+                if ($type = $p->getType()) {
+                    $k = $this->generateTypeHint($type, $r).' '.$k;
+                }
+            } elseif (preg_match('/^(?:[^ ]++ ){4}([a-zA-Z_\x7F-\xFF][^ ]++)/', $p, $type)) {
+                $k = $type[1].' '.$k;
+            }
+            if ($type && $p->allowsNull()) {
+                $k = '?'.$k;
+            }
+
+            try {
+                $k .= ' = '.$this->dumpValue($p->getDefaultValue(), false);
+                if ($type && $p->allowsNull() && null === $p->getDefaultValue()) {
+                    $k = substr($k, 1);
+                }
+            } catch (\ReflectionException $e) {
+                if ($type && $p->allowsNull() && !class_exists('ReflectionNamedType', false)) {
+                    $k .= ' = null';
+                    $k = substr($k, 1);
+                }
+            }
+
+            $signature[] = $k;
+        }
+
+        $signature = ($r->returnsReference() ? '&(' : '(').implode(', ', $signature).')';
+
+        if (method_exists($r, 'getReturnType') && $type = $r->getReturnType()) {
+            $signature .= ': '.($type->allowsNull() ? '?' : '').$this->generateTypeHint($type, $r);
```

Comment:
hmm, I forgot that PHP 7.0 has the method ``allowsNull`` even though return types cannot return null. But it is indeed logical as the same class is used for arguments too